### PR TITLE
Format Android lint violation code correctly in local report

### LINKED ISSUE #74: Parse and linkify URLs in Android lint comments
Example comment for an AllowBackup violation:

> Notes: On SDK version 23 and up, your app data will be automatically backed up and restored on app install. Consider adding the attribute `android:fullBackupContent` to specify an `@xml` resource which configures which files to backup. More info: https://developer.android.com/preview/backup/index.html

The URL is not linkified in reports. It could be!

### LINKED ISSUE #75: Format code in Android lint comments
Example comment for an AllowBackup violation:

> Notes: On SDK version 23 and up, your app data will be automatically backed up and restored on app install. Consider adding the attribute `android:fullBackupContent` to specify an `@xml` resource which configures which files to backup. More info: https://developer.android.com/preview/backup/index.html

The backticked code is not formatted using `pre` tags in reports. It could be!

### PR DESCRIPTION
**Pull request by stkent**
Closes #74 #75.

**Before (see `AllowBackup` violation):**

screen shot 2016-07-03 at 8 15 59 pm

**After (see `AllowBackup` violation):**

screen shot 2016-07-03 at 9 19 49 pm

---

I haven't tested this with the report task yet - need to check that HTML encoding is correct, etc.

### CODE REVIEW

File: plugin/src/main/groovy/com/btkelly/gnag/reporters/AndroidLintViolationDetector.groovy [hunk 1]
```diff
@@ -70,6 +70,8 @@ class AndroidLintViolationDetector extends BaseViolationDetector {
                 final String notNullMessageInHtml =
                         sanitizeToNonNull(markdownProcessor.markdownToHtml(messageInMarkdown))
                         .replaceAll("</?p>", "")
+
+            println markdownProcessor.markdownToHtml("https://developer.android.com/preview/backup/index.html")
```
Comment (stkent) [inline plugin/src/main/groovy/com/btkelly/gnag/reporters/AndroidLintViolationDetector.groovy]: Revert

**Comment by stkent** [comment]
Looks good when reported too, no regressions there: https://github.com/stkent/gnag-testing/pull/2#issuecomment-230483971

Though I did start thinking that maybe it would be better to format for local violations only... opinion?

**Comment by btkelly** [comment]
I mean formatting for local makes sense because GitHub will automatically create links for us but if it looks the same for remote comments I would prefer code simplicity. Let it format the same for both.

**Comment by btkelly** [comment]
Is the Android Linter the only one that has inline links like this? Just wondering if we should make this change for all of them.

**Comment by stkent** [comment]
As far as I know, yes, lint is the only tool that includes markdown and inline links in the message.

**Pull request merged**